tests/integration/components/flexberry-dropdown-test.js

```diff
@@ -0,0 +1,612 @@
+    value=value
+    items=itemsArray
+    needChecksOnValue=needChecksOnValue
+  }}`);
+
+  assert.strictEqual(this.get('value'), null, 'Component\'s property binded to \'value\' is equals to null');
+
+  // Change property binded to 'value' & check them.
+  this.set('needChecksOnValue', true);
+  let newValue = 'Caption4';
+  let latestLoggerErrorMessage;
+  Ember.Logger.error = function(message) {
+    latestLoggerErrorMessage = message;
+  };
+
+  // Check that errors handled properly.
+  this.set('value', newValue);
+  assert.strictEqual(Ember.typeOf(latestLoggerErrorMessage) === 'string', true, 'Check message exists');
+  assert.strictEqual(latestLoggerErrorMessage.indexOf(newValue) > 0, true, 'Invalide value exists');
+});
+
+test('itemsObject render properly', function(assert) {
+  assert.expect(4);
+
+  // Create objects for testing.
+  let itemsObject = {
+    item1: 'Caption1',
+    item2: 'Caption2',
+    item3: 'Caption3'
+  };
+  this.set('itemsObject', itemsObject);
+
+  // Render component.
+  this.render(hbs`{{flexberry-dropdown
+    items=itemsObject
+  }}`);
+
+  // Retrieve component.
+  let $component = this.$().children();
+  let $dropdownMenu = $component.children('div.menu');
+  let $dropdomnItem = $dropdownMenu.children('div.item');
+
+  // Check that  class 'item' in <menu> exist.
+  assert.strictEqual($dropdomnItem.hasClass('item'), true, 'Component\'s has class');
+
+  // Check component's captions and objects.
+  let itemsObjectKeys = Object.keys(itemsObject);
+  $dropdomnItem.each(function(i) {
+    let $item = Ember.$(this);
+    let itemKey = itemsObjectKeys[i];
+    let itemCaption = itemsObject[itemKey];
+
+    // Check that the captions matches the objects.
+    assert.strictEqual($item.attr('data-value'), itemCaption, 'Component\'s item\'s сaptions matches the objects');
+  });
+});
+
+test('itemsArray render properly', function(assert) {
+  assert.expect(4);
+
+  // Create array for testing.
+  let itemsArray = ['Caption1', 'Caption2', 'Caption3'];
+  this.set('itemsArray', itemsArray);
+
+  // Render component.
+  this.render(hbs`{{flexberry-dropdown
+    items=itemsArray
+  }}`);
+
+  // Retrieve component.
+  let $component = this.$().children();
+  let $dropdownMenu = $component.children('div.menu');
+  let $dropdomnItem = $dropdownMenu.children('div.item');
+
+  // Check that  class 'item' in <menu> exist.
+  assert.strictEqual($dropdomnItem.hasClass('item'), true, 'Component\'s has class');
+
+  // Check component's captions and array.
+  $dropdomnItem.each(function(i) {
+    let $item = Ember.$(this);
+    let itemCaption = itemsArray[i];
+
+    // Check that the captions matches the array.
+    assert.strictEqual($item.attr('data-value'), itemCaption, 'Component\'s item\'s сaptions matches the array');
+  });
+});
+
+// Call animation for component & check that animation handled properly.
+function clickOnDropdown(assert, resolve, $dropdomnItem, $component, $dropdownMenu) {
```

Comment:
All helper methods must be defined inthe beginning of the file, to avaid mixing of tests & helping methods.